plugins/woocommerce/src/Internal/Admin/Translations.php

```diff
@@ -93,11 +93,17 @@ private function get_translation_chunk_data( $lang_dir, $domain, $locale ) {
 				continue;
 			}
 
-			if ( ! isset( $chunk_data['comment']['reference'] ) ) {
+			// Support for wp-cli --make-json and Loco Translate file format.
+
+			if ( ! ( isset( $chunk_data['comment']['reference'] ) || isset( $chunk_data['source'] ) ) ) {
```

Comment:
I think it would be better extracting each parser into separate functions rather than lumping them into one. For me, the resulting nested `if` branch conditions add too much complexity to this function.

An alternative might be to have `get_translation_chunk_data`, perform heuristics on the file to detect the format, and then call the appropriate parser function for that format.

Reply:
Hi @kenarsuleyman, Would you be able to make a change as Adrian's suggestion? 🙏

Reply:
Hi, @chihsuan since this is only a workaround fix and will get reverted sometime in near future, I might refactor the entire function based on Adrian's suggestion. Right now I don't know how long this fix needs to stay in the codebase.

Reply:
> If current version is not suitable for standarts to merge it, I can do it this week.

Hi @kenarsuleyman Yeah, I think that would be perfect, thanks!

Reply:
Hi @adrianduffell, I wrote separate functions to combine translation chunks after your suggestion. I agree that this made the code more readable only problem is I'm not fully happy how those separate functions ended up with. There are "duplicate looking" lines, they are not exact duplicates, just the same logic for different file formats. Unfortunately, I couldn't think of a more optimized approach.

Reply:
Hi @kenarsuleyman Just saw you push a new commit, we will review it soon. 👍  Thank you!

Sorry that @adrianduffell is unavailable.